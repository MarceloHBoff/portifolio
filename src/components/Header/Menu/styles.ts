import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.div<{ isOpen: boolean }>`
  ${({ theme, isOpen }) => css`
    display: ${isOpen ? 'flex' : 'none'};
    justify-content: space-around;
    align-items: center;

    ${isOpen &&
    media.lessThan('large')`
      width: 100%;
      position: fixed;
      top: 0;
      bottom: 0;
      right: 0;
      left: 0;
      background: ${theme.colors.background};
      z-index: 1;

      display: flex;
      flex-direction: column;
      align-items: center;
    `}
  `}
`

export const MenuItem = styled.button<{ selected: boolean }>`
  ${({ theme, selected }) => css`
    font-size: ${theme.font.sizes.xlarge};
    color: ${theme.colors.text};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};

    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0;
      border-radius: ${theme.border.radius};
      border-bottom: ${theme.border.big} solid ${theme.colors.text};
      transition: transform 0.3s ease-in-out;
      transform: scaleX(0);
    }

    &:hover::after {
      transform: scaleX(1);
    }

    ${selected &&
    css`
      &::after {
        transform: scaleX(1);
        border-color: ${theme.colors.primary};
      }
    `}
  `}
`
