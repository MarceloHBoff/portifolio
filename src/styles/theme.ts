export default {
  colors: {
    background: '#283747',
    text: '#FAFAFA',
    textBlack: '#030517',
    header: '#17202A',
    primary: '#EB984E',
    primaryDark: '#C85E00',
    secundary: '#222',
    error: '#c53030'
  },

  font: {
    family: 'Roboto, sans-serif',
    light: 300,
    normal: 400,
    bold: 700,

    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem'
    }
  },

  border: {
    radius: '1rem',
    small: '0.1rem',
    big: '0.2rem'
  },

  spacings: {
    xsmall: '0.5rem',
    small: '1rem',
    normal: '1.5rem',
    large: '2rem',
    xlarge: '2.5rem',
    huge: '3rem',
    xhuge: '4rem'
  }
} as const
