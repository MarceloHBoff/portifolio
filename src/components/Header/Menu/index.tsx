import { NextRouter } from 'next/dist/client/router'
import Link from 'next/link'

import { useCallback } from 'react'

import { Container, MenuItem } from './styles'

type MenuProps = {
  isOpen?: boolean
  setIsOpen?: (isOpen: boolean) => void
  router: NextRouter
}

export default function Menu({ isOpen = true, setIsOpen, router }: MenuProps) {
  const handleClick = useCallback(() => {
    if (setIsOpen) setIsOpen(false)
  }, [setIsOpen])

  return (
    <Container isOpen={isOpen} onClick={handleClick}>
      <Link href="/">
        <MenuItem selected={router.pathname === '/'}>Projetos</MenuItem>
      </Link>
      <Link href="/whoIAm">
        <MenuItem selected={router.pathname === '/whoIAm'}>Sobre</MenuItem>
      </Link>
      <Link href="/contact">
        <MenuItem selected={router.pathname === '/contact'}>Contato</MenuItem>
      </Link>
    </Container>
  )
}
