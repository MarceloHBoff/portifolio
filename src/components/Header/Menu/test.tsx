import { NextRouter } from 'next/dist/client/router'

import { screen } from '@testing-library/react'

import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import Menu from '.'

describe('<Menu />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(
      <Menu router={{ pathname: '/' } as NextRouter} />
    )

    const projetos = screen.getByText(/projetos/i)
    expect(projetos).toBeInTheDocument()
    expect(screen.getByText(/sobre/i)).toBeInTheDocument()
    expect(screen.getByText(/contato/i)).toBeInTheDocument()

    expect(projetos).toHaveStyleRule('border-color', theme.colors.primary, {
      modifier: '::after'
    })

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render selected a specefic menu', () => {
    renderWithTheme(<Menu router={{ pathname: '/contact' } as NextRouter} />)

    expect(screen.getByText(/contato/i)).toHaveStyleRule(
      'border-color',
      theme.colors.primary,
      {
        modifier: '::after'
      }
    )
  })
})
