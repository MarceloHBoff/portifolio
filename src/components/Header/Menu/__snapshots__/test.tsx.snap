// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 2.0rem;
  color: #FAFAFA;
  padding: 0.5rem 1rem;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 1rem;
  border-bottom: 0.3rem solid #FAFAFA;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
}

.c1:hover::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c1::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
  border-color: #A6DAF7;
}

.c2 {
  font-size: 2.0rem;
  color: #FAFAFA;
  padding: 0.5rem 1rem;
  position: relative;
}

.c2::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 1rem;
  border-bottom: 0.3rem solid #FAFAFA;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
}

.c2:hover::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

@media (max-width:1170px) {
  .c0 {
    width: 100%;
    position: fixed;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    background: #121212;
    z-index: 1;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<body>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        Projetos
      </button>
      <button
        class="c2"
      >
        Sobre
      </button>
      <button
        class="c2"
      >
        Contato
      </button>
    </div>
  </div>
</body>
`;
